src/Microsoft.AspNet.Mvc.Core/ModelBinding/Validation/DefaultComplexObjectValidationStrategy.cs

```diff
@@ -78,7 +81,34 @@ public bool MoveNext()
                 var property = _properties[_index];
                 var propertyName = property.BinderModelName ?? property.PropertyName;
                 var key = ModelNames.CreatePropertyModelName(_key, propertyName);
-                var model = property.PropertyGetter(_model);
+
+                object model;
+
+                // Our property accessors don't work on Mono 4.0.4 - see https://bugzilla.xamarin.com/show_bug.cgi?id=25717
+                // This is a workaround for what the PropertyGetter does in the background.
+                if (IsMono)
+                {
+                    if (_model == null)
+                    {
+                        model = null;
+                    }
+                    else
+                    {
+                        var propertyInfo = _model.GetType().GetRuntimeProperty(property.PropertyName);
```

Comment:
Note that neither version of this code gracefully handles set-only properties. Will address separately.